src/test/java/uk/gov/ons/census/casesvc/service/UacProcessorTest.java

```diff
@@ -76,4 +88,94 @@ public void testEmitUacUpdatedEvent() {
+        uacQidLinkArgumentCaptor.getValue().getUac());
+    assertEquals(
+        uacCreatedEvent.getPayload().getUacQidCreated().getCaseId(),
+        uacQidLinkArgumentCaptor.getValue().getCaze().getCaseId());
+  }
+
+  @Test
+  public void testIngestUacCreatedEventEmitsUacUpdatedEvent() {
+    // Given
+    Case linkedCase = getRandomCase();
+    ResponseManagementEvent uacCreatedEvent = generateUacCreatedEvent(linkedCase);
+    when(caseRepository.findByCaseId(uacCreatedEvent.getPayload().getUacQidCreated().getCaseId()))
+        .thenReturn(Optional.of(linkedCase));
+    ArgumentCaptor<ResponseManagementEvent> responseManagementEventArgumentCaptor =
+        ArgumentCaptor.forClass(ResponseManagementEvent.class);
+    ReflectionTestUtils.setField(underTest, "outboundExchange", "TEST_EXCHANGE");
+
+    // When
+    underTest.ingestUacCreatedEvent(uacCreatedEvent);
+
+    // Then
+    verify(rabbitTemplate)
+        .convertAndSend(
+            eq("TEST_EXCHANGE"),
+            eq("event.uac.update"),
+            responseManagementEventArgumentCaptor.capture());
+    assertEquals(
+        uacCreatedEvent.getPayload().getUacQidCreated().getQid(),
+        responseManagementEventArgumentCaptor
+            .getValue()
+            .getPayload()
+            .getUac()
+            .getQuestionnaireId());
+    assertEquals(
+        uacCreatedEvent.getPayload().getUacQidCreated().getUac(),
+        responseManagementEventArgumentCaptor.getValue().getPayload().getUac().getUac());
+    assertEquals(
+        uacCreatedEvent.getPayload().getUacQidCreated().getCaseId().toString(),
+        responseManagementEventArgumentCaptor.getValue().getPayload().getUac().getCaseId());
+  }
+
+  @Test
+  public void testIngestUacCreatedEventLogsRmUacCreatedEvent() {
+    // Given
+    Case linkedCase = getRandomCase();
+    ResponseManagementEvent uacCreatedEvent = generateUacCreatedEvent(linkedCase);
+    when(caseRepository.findByCaseId(uacCreatedEvent.getPayload().getUacQidCreated().getCaseId()))
+        .thenReturn(Optional.of(linkedCase));
+    ArgumentCaptor<UacQidLink> uacQidLinkArgumentCaptor = ArgumentCaptor.forClass(UacQidLink.class);
+
+    // When
+    underTest.ingestUacCreatedEvent(uacCreatedEvent);
+
+    // Then
+    verify(eventLogger)
+        .logEvent(
+            uacQidLinkArgumentCaptor.capture(),
+            eq("RM UAC QID pair created"),
+            eq(uacCreatedEvent.getPayload()),
+            eq(uacCreatedEvent.getEvent()));
+    assertEquals(
+        uacCreatedEvent.getPayload().getUacQidCreated().getQid(),
+        uacQidLinkArgumentCaptor.getValue().getQid());
+    assertEquals(
+        uacCreatedEvent.getPayload().getUacQidCreated().getUac(),
+        uacQidLinkArgumentCaptor.getValue().getUac());
+    assertEquals(
+        uacCreatedEvent.getPayload().getUacQidCreated().getCaseId(),
+        uacQidLinkArgumentCaptor.getValue().getCaze().getCaseId());
+  }
 }
```

Comment:
There should be some sad :( path tests.  Checking that a RunTimeException is thrown when linkedCase.isEmpty() for example.  There's plenty of example in the project.   e.g. testReceiptedQidNotFound.  Again this isn't consistent across even this project